test/e2e/e2e_test.go

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package e2e
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"os/exec"
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+// Test context
+var (
+	ctx context.Context
+)
+
+func TestMain(t *testing.T) {
+	suiteName := "E2E NVIDIA Container Toolkit"
+
+	RegisterFailHandler(Fail)
+	RunSpecs(t,
+		suiteName,
+	)
+}
+
+// BeforeSuite runs before the test suite
+var _ = BeforeSuite(func() {
+	ctx = context.Background()
+
+	// Pull the images to be used in the tests
+	// this cleans the stdout comming from the docker run command
+	_, err := runScript("docker pull ubuntu")
```

Comment:
This implies that these tests will only ever run docker tests. It's not a blocker for this PR, but we should at least consider how we may want to handle other runtimes (e.g. podman).

Reply:
right now we can leave the Makefile as is, moving forward we can control the runtime to be tested with flags/env variables on the makefile